src/t8_forest/t8_forest_vtk.cxx

```diff
@@ -418,12 +423,26 @@ for (int idata = 0; idata < num_data; idata++) {
 
 T8_FREE (cellTypes);
 T8_FREE (dataArrays);
+/* Writing was successful */
+return 1;
+
+t8_forest_vtk_failure:
+t8_errorf ("Error when writing vtk file.\n");
+vtk_treeid->Delete ();
+vtk_mpirank->Delete ();
+vtk_level->Delete ();
+vtk_element_id->Delete ();
+for (int idata = 0; idata < num_data; idata++) {
+  dataArrays[idata]->Delete ();
+}
```

Comment:
See comment above, remove the duplicated code.
The error print is good and should be triggered if the return value of write was 0.